### ISSUE TITLE
Buttons that have visible text should not use tooltips - New occurrences

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55815]
### Description

Previously: https://github.com/WordPress/gutenberg/issues/42676 / https://github.com/WordPress/gutenberg/pull/42815

This was already fixed for the buttons in the top bar in https://github.com/WordPress/gutenberg/pull/42815 but it appears it's something that many contributors miss. I spotted a few new occurrences of this issue in newsly / refactored buttons.
It would be great to consider to automate this behavior in the components so that we don't have to worry about it again. Cc @ciampo 

- Buttons that have visible text should not use tooltips to repeat the visible text.
- The only case where a tooltip should be used for buttons with visible text is when the button action is associated with a keyboard shortcut. So far, tooltips are the only place where the keyboard shortcut is visually exposed, besides the dedicated modal dialog. Although that's not ideal, it's the design convention used in the editor so far.

### Step-by-step reproduction instructions

- Edit a post that is a draft.
- Use the keyboard to navigate the top bar buttons.
- Move focus to the 'Save' button.
- Observe the button does have visible text.
- Observe the button does show a tooltip that just repeats the visible text. Screenshot:

<img width="382" alt="Screenshot 2023-09-19 at 09 37 07" src="https://github.com/WordPress/gutenberg/assets/1682452/eb18c8d5-b9e1-42d6-be21-b3c748e4f475">

- Additionally: Observe the button doesn't have a focus style. Worth reminding we keep 'disabled' (actually aria-disabled) buttons focusable for discoverability and normally they show a ligher focus style. E.g. the 'disabled' Undo button. Screenshot:

![focus indication disabled control](https://github.com/WordPress/gutenberg/assets/1682452/926d805e-ce0b-44ac-b1d5-15956f7c246a)

- Move focus to the Preview button. Since in the default state it's an icon button, it's okay to show a tooltip. Screenshot:

![Screenshot 2023-11-02 at 15 27 30](https://github.com/WordPress/gutenberg/assets/1682452/88ea5a3b-13b1-4079-9c40-894a95f726ed)

- Go to Options > Preferences, and enable 'Show button text labels'.
- The Preview button now shows visible text instead of an icon.
- Move focus to the Preview button again.
- Observe the button shows a tooltip that just repeates the visible text. Screenshot:
- 
![Screenshot 2023-11-02 at 15 27 52](https://github.com/WordPress/gutenberg/assets/1682452/b5aaac55-111f-4c25-bfce-6d319788202d)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
The 'Save/Saved' button has its own specific issues previously reported on https://github.com/WordPress/gutenberg/pull/43952 and https://github.com/WordPress/gutenberg/discussions/44735

I will keep it out for now.

So far, we are forcinc the tooltip of this button to always render, to avoid a focus loss. This is very specific to this button, because rendering the tooltip conditionally based on the button state would trigger a full re-rendering of the DOM element and thus trigger a focus loss. It is still possible to reproduce the potential focus loss [by removing this showTooltip prop](https://github.com/WordPress/gutenberg/blob/621d2a2a0e7ed221b8820776a1f664349cbbcce3/packages/editor/src/components/post-saved-state/index.js#L161) and then saving a post edit by pressing the Save button with the keyboard.
As such, for now we have to keep forcing the tooltip on the Save button, even though it's not ideal.

Looks like the refactoring of the Tooltip with using ariakit didn't adress this issue. @ciampo @talldan could you please confirm when you have a chance?

**Comment by ciampo** [comment]
> It would be great to consider to automate this behavior in the components so that we don't have to worry about it again

I'm not sure we can find a way for the `Button` component to reliably understand if its `children` contain _visible_ text. I am also afraid that this change may introduce unexpected changes across the many usages of the component.
My preference would be to continue to rely on consumers of the component to correctly show tooltips when needed. I wonder if we can instead avoid such regressions with automated testing.

> It is still possible to reproduce the potential focus loss [by removing this showTooltip prop](https://github.com/WordPress/gutenberg/blob/621d2a2a0e7ed221b8820776a1f664349cbbcce3/packages/editor/src/components/post-saved-state/index.js#L161) and then saving a post edit by pressing the Save button with the keyboard.
As such, for now we have to keep forcing the tooltip on the Save button, even though it's not ideal.
> Looks like the refactoring of the Tooltip with using ariakit didn't adress this issue.

I gave this a quick look and the issue doesn't seem related to the `Tooltip` component anymore, but more to the fact that [the `Button` component conditionally renders a component around `{element}` based on the `shouldShowTooltip` variable](https://github.com/WordPress/gutenberg/blob/aa7cc4d0404e0a830144c57e5211335b48f9ba2d/packages/components/src/button/index.tsx#L259-L285). The same reconciliation problem would happen if we rendered a `div` instead of the `Tooltip` component.

**Comment by afercia** [comment]
> > It would be great to consider to automate this behavior in the components so that we don't have to worry about it again
> 
> I'm not sure we can find a way for the `Button` component to reliably understand if its `children` contain _visible_ text. I am also afraid that this change may introduce unexpected changes across the many usages of the component. My preference would be to continue to rely on consumers of the component to correctly show tooltips when needed. I wonder if we can instead avoid

@ciampo thanks for your feedback. I'm not sure I understand what the action plan is here, your comment seems truncated at the end :)

**Comment by ciampo** [comment]
> @ciampo thanks for your feedback. I'm not sure I understand what the action plan is here, your comment seems truncated at the end :)

Woops, I must have completed the sentences only in my head 🙈  I updated my original comment, thank you for spotting it!
